microcosm_elasticsearch/searching.py

```diff
@@ -138,15 +134,7 @@ def _query(self):
         Starts with the index's search function; customizes with the provided doc types, if any
 
         """
-        query = self.index.search()
-        query = self._filter_terms(query)
-        return query
-
-    def _filter_terms(self, query):
-        return query.filter(
```

Comment:
We were overriding this default doc type limiting filter in all of our actual usages. So I'm removing `_filter_terms` from here and adding some extra, optional support in `_filter`